hazelcast/src/main/java/com/hazelcast/collection/impl/list/ListContainer.java

```diff
@@ -166,7 +166,9 @@ public int indexOf(boolean last, Data value) {
         if (itemList == null) {
             if (itemMap != null && !itemMap.isEmpty()) {
                 itemList = new ArrayList<CollectionItem>(itemMap.values());
-                Collections.sort(itemList);
+                sort(itemList);
+                CollectionItem lastItem = itemList.get(itemList.size() - 1);
```

Comment:
This is a fix comparable to https://github.com/hazelcast/hazelcast/pull/2844 and https://github.com/hazelcast/hazelcast/pull/4275. The sort was done before, so it's cheap to retrieve the item with the greatest `itemId` (if I interpreted the `Comparable` implementation of `CollectionItem` correctly).